sympy/core/containers.py

```diff
@@ -50,6 +50,12 @@ class Tuple(Basic):
     def __new__(cls, *args, **kwargs):
         if kwargs.get('sympify', True):
             args = (sympify(arg) for arg in args)
+        #if kwargs.get('denest_lists', False):
+        #    args = (Tuple(*arg) if isinstance(arg, list) else arg for arg in args)
+        args = tuple(args)
+        for arg in args:
+            if isinstance(arg, list):
+                warnings.warn(DeprecationWarning('Tuple',arg))
```

Comment:
Deprecation warnings in SymPy are usually given like this:
https://github.com/sympy/sympy/blob/ed0bfd1e47345c8bd45ef542eef18f6a21c55685/sympy/core/sympify.py#L453-L460
The SymPyDeprecationWarning class is handled specially by the test runner so that it is an error if it is emitted in the test suite.

Reply:
As far as I am concerned, this warning will be removed before the merge (see also the first review comment by me). If such a warning is to be added, it should be added to `Basic` when all such issues are resolved. This is for demonstration purposes. (that there are no longer such violations)